src/App.tsx

```diff
@@ -27,47 +24,63 @@ function App() {
   const [showLegends, setShowLegends] = useState(true);
   const [pointCount, setPointCount] = useState(0);
   const [mapView, setMapView] = useState<__esri.MapView>();
-  // fetch the data from a MockData.json
-  if (mapView) {
-    // Verify the map is modified
-    fetch("./MockData.json").then((data) => {
-      console.log("Data from MockData.json:", data);
-    });
+  /*
+We used useEffect here to ensure that once the data is successfully loaded from the mapView, 
+it triggers the fetching of data from the JSON file and then displays that data in the console.
+*/
+  useEffect(() => {
+    const fetchData = async () => {
+      if (mapView) {
+        try {
+          // Fetch data from the local JSON file
+          const response = await fetch("./MockData.json");
+          /*
+          We used the await keyword here to ensure that the operation does not pause or delay while waiting for data
+           to be fetched from the JSON file. Instead, it allows the code to continue running asynchronously
+            until the data is fully retrieved and available for use. This way, the data appears without any
+             unnecessary delays.
```

Comment:
We used the await keyword here to ensure that the operation does not pause or delay while waiting for data
           to be fetched from the JSON file. Instead, it allows the code to continue running asynchronously
            until the data is fully retrieved and available for use. This way, the data appears without any
             unnecessary delays.